versioned/persist/store/src/main/java/org/projectnessie/versioned/persist/store/PersistVersionStore.java

```diff
@@ -135,8 +138,26 @@ public Hash commit(
           } else {
             expectedValue = Optional.empty();
           }
-          commitAttempt.putExpectedStates(contentId, expectedValue);
-          commitAttempt.putGlobal(contentId, newState);
+
+          // Handle the case when there are multiple operations against the same contents-id.
+          if (!expectedStates.containsKey(contentId)) {
+            // First occurrence of contentsId in this commit.
+            commitAttempt.putExpectedStates(contentId, expectedValue);
+            expectedStates.put(contentId, expectedValue);
+          } else {
+            // Consecutive occurrence of contentsId in this commit.
+            if (expectedValue.isPresent()) {
+              // Operation expects a certain global, compare against the previous global-value
+              // from this commit.
+              if (!globals.get(contentId).equals(expectedValue.get())) {
```

Comment:
According to `spec.md` I think we should deny this `Put` even if the global state matches, because the spec requires at most one `Put` per key... WDYT?

Reply:
sgtm

Reply:
I confused keys with IDs, here, sorry... closing this conversation for now.